<launch>
  <node name="finish_line_watcher" pkg="rr_iarrc" type="finish_line_watcher" output="screen">
      <param name="img_topic" value="/finish_line_detection_img" />

      <!-- Params to control blocking of the image. (Image Height = 1280)-->
      <!-- Sky height draws the a block from y=0 to y=sky height (increasing sky height lowers the sky rectangle) -->
      <param name="blockSky_height" type="int" value="560" />
      <!-- Wheels height draws the a block from the bottom to y=wheel height (increasing wheel height brings the rectangle up bigger from the bottom) -->
      <param name="blockWheels_height" type="int" value="750" />
      <!-- Same thing as wheel height, just draws the rectangle over the middle third of the screen -->
      <param name="blockBumper_height" type="int" value="720" />

      <!--If set to true, the detector will start publishing once the finish line is first detected, if false, publishes when we cross the finish line-->
      <param name="publish_when_detected" type="boolean" value="false" />

      <!--The detector uses line detection and will only consider relatively flat lines (slope < slope cutoff) and will only trigger when relatively long lines are detected (length > length_cutoff) -->
      <param name="slope_cutoff" type="double" value="0.3" />
      <param name="length_cutoff" type="double" value="75.0" />

      <!-- Number of camera frames between detections. Added to prevent flickering and double detections. -->
      <param name="cooldown" type="int" value="7" />

      <!-- As a final sanity check the detector counts non-zero pixels in the HSV threshold. Count of non-zero pixels must be > this threshold for the detector to fire. -->
      <param name="count_thresh" type="int" value="2000" />

      <!-- Some params for Canny edge detection and Hough line detection. Refer to OpenCV docs for meaning -->
      <param name="canny_thresh1" type="double" value="50.0" />
      <param name="canny_thresh2" type="double" value="200.0" />
      <param name="hough_rho" type="double" value="1.0" />
      <param name="hough_theta" type="double" value="0.01745327777" /> <!-- pi/180 -->
      <param name="hough_thresh" type="int" value="50" />
      <param name="hough_minLineLength" type="double" value="50" />
      <param name="hough_maxLineGap" type="double" value="35" />
  </node>

  <node pkg="nodelet" type="nodelet" name="color_detector_finish_line" output="screen"
        args="standalone rr_iarrc/color_detector">
      <param name="load_file" value= "$(find rr_iarrc)/saved_hsv/finish_line_magenta.txt"/>
      <param name="camera_image" value="/camera_center/image_color_rect"/>
      <param name="detection_topic" value="/finish_line_detection_img"/>
      <param name="hsv_values_topic" value="/finish_line_hsv"/>
  </node>

  <node name="hsv_tuner" pkg="rr_iarrc" type="hsv_tuner" output="screen">
      <param name="load_file" value= "$(find rr_iarrc)/saved_hsv/finish_line_magenta.txt"/>
      <param name="values_topic" value="/finish_line_hsv"/>
  </node>
</launch>
